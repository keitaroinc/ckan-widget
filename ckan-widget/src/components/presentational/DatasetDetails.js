import React, { Component } from 'react'
import MaterialIcon from 'material-icons-react'

class DatasetDetails extends Component {
    renderResources = (resources, datasetName) => {
        let items = []
        resources.forEach((resource, i) => {
            let restricted = resource.restricted ? JSON.parse(resource.restricted) : 'N/A'
            items.push(
                <li className="list-group-item d-flex flex-wrap align-items-center" key={i}>
                    <span className="type">
                        <span className="badge badge-secondary">{resource.format}</span>
                    </span>
                    <a className="px-3 title" href={resource.url}>{resource.name}</a>
                    <span className="ml-auto mr-3 date">{this.formatDate(resource.last_modified)}</span>
                    <span className="visibility">
                        <span className="badge badge-pill badge-dark">{restricted.level}</span>
                    </span>
                </li>
            )
        })

        return items
    }

    formatDate = date => {
        const d = new Date(date)
        return d.toUTCString()
    }

    render() {
        const {
            name,
            notes,
            resources,
            dataset_creation_date,
            dataset_modification_date,
            dataset_publication_date,
            organization,
            collapsed
        } = this.props
        const orgName = organization !== null ? organization.name : 'N/A'
        const collapseClass = collapsed ? 'collapse' : ''

        return(
            <div className={"card-footer px-5 py-4 " + collapseClass}>
                <p className="lead">{notes}</p>
                <hr/>
                <ul className="text-muted list-inline">
                    <li className="list-inline-item"><strong>Created on:</strong> {dataset_creation_date}</li>
                    <li className="list-inline-item"><strong>Publication date:</strong> {dataset_publication_date}</li>
                    <li className="list-inline-item"><strong>Modification date:</strong> {dataset_modification_date}</li>
                    <li className="list-inline-item"><strong>Organization:</strong> {orgName}</li>
                </ul>
                <div className="my-4">
                    <h3>Resources</h3>
                    <ul className="list-inline">
                        { this.renderResources(resources, name) }
                    </ul>
                </div>
                <a className="btn btn-success mb-1" href={`https://trouver.datasud.fr/dataset/${name}`}>
                    <MaterialIcon icon="open_in_new" size="tiny" color="#fff" />
                    <span className="ml-1">View on Datasud.fr</span>
                </a>
            </div>
        )
    }
}

export default DatasetDetails
