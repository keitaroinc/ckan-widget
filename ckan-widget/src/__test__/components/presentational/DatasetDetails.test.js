import React from 'react';
import { shallow } from 'enzyme';
import DatasetDetails from '../../../components/presentational/DatasetDetails';

describe('DatasetDetails', () => {
  it('should be defined', () => {
    expect(DatasetDetails).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = shallow(
      <DatasetDetails organization={{ name: 'org1' }} resources={[{ restricted: "{\"level\": \"public\"}", format: 'CSV', name: 'rsc1', last_modified: '21 June 2018' }]}  />
    );
    expect(tree).toMatchSnapshot();
  });
});