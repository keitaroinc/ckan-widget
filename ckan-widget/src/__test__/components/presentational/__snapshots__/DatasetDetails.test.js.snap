// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetDetails should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DatasetDetails
    organization={
        Object {
            "name": "org1",
          }
    }
    resources={
        Array [
            Object {
              "format": "CSV",
              "last_modified": "21 June 2018",
              "name": "rsc1",
              "restricted": "{\\"level\\": \\"public\\"}",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="lead"
/>,
        <hr />,
        <ul
          className="text-muted list-inline"
>
          <li
                    className="list-inline-item"
          >
                    <strong>
                              Created on:
                    </strong>
                     
          </li>
          <li
                    className="list-inline-item"
          >
                    <strong>
                              Publication date:
                    </strong>
                     
          </li>
          <li
                    className="list-inline-item"
          >
                    <strong>
                              Modification date:
                    </strong>
                     
          </li>
          <li
                    className="list-inline-item"
          >
                    <strong>
                              Organization:
                    </strong>
                     
                    org1
          </li>
</ul>,
        <div
          className="my-4"
>
          <h3>
                    Resources
          </h3>
          <ul
                    className="list-inline"
          >
                    <li
                              className="list-group-item d-flex align-items-center"
                    >
                              <span
                                        className="badge badge-secondary"
                              >
                                        CSV
                              </span>
                              <a
                                        className="px-3"
                                        href={undefined}
                              >
                                        rsc1
                              </a>
                              <span
                                        className="ml-auto mr-3"
                              >
                                        Wed, 20 Jun 2018 22:00:00 GMT
                              </span>
                              <span
                                        className="badge badge-pill badge-dark"
                              >
                                        public
                              </span>
                    </li>
          </ul>
</div>,
        <a
          className="btn btn-success mb-1"
          href="https://trouver.datasud.fr/dataset/undefined"
>
          <i
                    className="material-icons mr-1"
          >
                    open_in_new
          </i>
           View on Datasud.fr
</a>,
      ],
      "className": "card-footer px-5 py-4 ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "lead",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="list-inline-item"
>
              <strong>
                            Created on:
              </strong>
               
</li>,
            <li
              className="list-inline-item"
>
              <strong>
                            Publication date:
              </strong>
               
</li>,
            <li
              className="list-inline-item"
>
              <strong>
                            Modification date:
              </strong>
               
</li>,
            <li
              className="list-inline-item"
>
              <strong>
                            Organization:
              </strong>
               
              org1
</li>,
          ],
          "className": "text-muted list-inline",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <strong>
                  Created on:
</strong>,
                " ",
                undefined,
              ],
              "className": "list-inline-item",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Created on:",
                },
                "ref": null,
                "rendered": "Created on:",
                "type": "strong",
              },
              " ",
              undefined,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <strong>
                  Publication date:
</strong>,
                " ",
                undefined,
              ],
              "className": "list-inline-item",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Publication date:",
                },
                "ref": null,
                "rendered": "Publication date:",
                "type": "strong",
              },
              " ",
              undefined,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <strong>
                  Modification date:
</strong>,
                " ",
                undefined,
              ],
              "className": "list-inline-item",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Modification date:",
                },
                "ref": null,
                "rendered": "Modification date:",
                "type": "strong",
              },
              " ",
              undefined,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <strong>
                  Organization:
</strong>,
                " ",
                "org1",
              ],
              "className": "list-inline-item",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Organization:",
                },
                "ref": null,
                "rendered": "Organization:",
                "type": "strong",
              },
              " ",
              "org1",
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Resources
</h3>,
            <ul
              className="list-inline"
>
              <li
                            className="list-group-item d-flex align-items-center"
              >
                            <span
                                          className="badge badge-secondary"
                            >
                                          CSV
                            </span>
                            <a
                                          className="px-3"
                                          href={undefined}
                            >
                                          rsc1
                            </a>
                            <span
                                          className="ml-auto mr-3"
                            >
                                          Wed, 20 Jun 2018 22:00:00 GMT
                            </span>
                            <span
                                          className="badge badge-pill badge-dark"
                            >
                                          public
                            </span>
              </li>
</ul>,
          ],
          "className": "my-4",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Resources",
            },
            "ref": null,
            "rendered": "Resources",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="list-group-item d-flex align-items-center"
>
                  <span
                                    className="badge badge-secondary"
                  >
                                    CSV
                  </span>
                  <a
                                    className="px-3"
                                    href={undefined}
                  >
                                    rsc1
                  </a>
                  <span
                                    className="ml-auto mr-3"
                  >
                                    Wed, 20 Jun 2018 22:00:00 GMT
                  </span>
                  <span
                                    className="badge badge-pill badge-dark"
                  >
                                    public
                  </span>
</li>,
              ],
              "className": "list-inline",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="badge badge-secondary"
>
                      CSV
</span>,
                    <a
                      className="px-3"
                      href={undefined}
>
                      rsc1
</a>,
                    <span
                      className="ml-auto mr-3"
>
                      Wed, 20 Jun 2018 22:00:00 GMT
</span>,
                    <span
                      className="badge badge-pill badge-dark"
>
                      public
</span>,
                  ],
                  "className": "list-group-item d-flex align-items-center",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "CSV",
                      "className": "badge badge-secondary",
                    },
                    "ref": null,
                    "rendered": "CSV",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "rsc1",
                      "className": "px-3",
                      "href": undefined,
                    },
                    "ref": null,
                    "rendered": "rsc1",
                    "type": "a",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Wed, 20 Jun 2018 22:00:00 GMT",
                      "className": "ml-auto mr-3",
                    },
                    "ref": null,
                    "rendered": "Wed, 20 Jun 2018 22:00:00 GMT",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "public",
                      "className": "badge badge-pill badge-dark",
                    },
                    "ref": null,
                    "rendered": "public",
                    "type": "span",
                  },
                ],
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="material-icons mr-1"
>
              open_in_new
</i>,
            " View on Datasud.fr",
          ],
          "className": "btn btn-success mb-1",
          "href": "https://trouver.datasud.fr/dataset/undefined",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "open_in_new",
              "className": "material-icons mr-1",
            },
            "ref": null,
            "rendered": "open_in_new",
            "type": "i",
          },
          " View on Datasud.fr",
        ],
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="lead"
/>,
          <hr />,
          <ul
            className="text-muted list-inline"
>
            <li
                        className="list-inline-item"
            >
                        <strong>
                                    Created on:
                        </strong>
                         
            </li>
            <li
                        className="list-inline-item"
            >
                        <strong>
                                    Publication date:
                        </strong>
                         
            </li>
            <li
                        className="list-inline-item"
            >
                        <strong>
                                    Modification date:
                        </strong>
                         
            </li>
            <li
                        className="list-inline-item"
            >
                        <strong>
                                    Organization:
                        </strong>
                         
                        org1
            </li>
</ul>,
          <div
            className="my-4"
>
            <h3>
                        Resources
            </h3>
            <ul
                        className="list-inline"
            >
                        <li
                                    className="list-group-item d-flex align-items-center"
                        >
                                    <span
                                                className="badge badge-secondary"
                                    >
                                                CSV
                                    </span>
                                    <a
                                                className="px-3"
                                                href={undefined}
                                    >
                                                rsc1
                                    </a>
                                    <span
                                                className="ml-auto mr-3"
                                    >
                                                Wed, 20 Jun 2018 22:00:00 GMT
                                    </span>
                                    <span
                                                className="badge badge-pill badge-dark"
                                    >
                                                public
                                    </span>
                        </li>
            </ul>
</div>,
          <a
            className="btn btn-success mb-1"
            href="https://trouver.datasud.fr/dataset/undefined"
>
            <i
                        className="material-icons mr-1"
            >
                        open_in_new
            </i>
             View on Datasud.fr
</a>,
        ],
        "className": "card-footer px-5 py-4 ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "lead",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="list-inline-item"
>
                <strong>
                                Created on:
                </strong>
                 
</li>,
              <li
                className="list-inline-item"
>
                <strong>
                                Publication date:
                </strong>
                 
</li>,
              <li
                className="list-inline-item"
>
                <strong>
                                Modification date:
                </strong>
                 
</li>,
              <li
                className="list-inline-item"
>
                <strong>
                                Organization:
                </strong>
                 
                org1
</li>,
            ],
            "className": "text-muted list-inline",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <strong>
                    Created on:
</strong>,
                  " ",
                  undefined,
                ],
                "className": "list-inline-item",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Created on:",
                  },
                  "ref": null,
                  "rendered": "Created on:",
                  "type": "strong",
                },
                " ",
                undefined,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <strong>
                    Publication date:
</strong>,
                  " ",
                  undefined,
                ],
                "className": "list-inline-item",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Publication date:",
                  },
                  "ref": null,
                  "rendered": "Publication date:",
                  "type": "strong",
                },
                " ",
                undefined,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <strong>
                    Modification date:
</strong>,
                  " ",
                  undefined,
                ],
                "className": "list-inline-item",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Modification date:",
                  },
                  "ref": null,
                  "rendered": "Modification date:",
                  "type": "strong",
                },
                " ",
                undefined,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <strong>
                    Organization:
</strong>,
                  " ",
                  "org1",
                ],
                "className": "list-inline-item",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Organization:",
                  },
                  "ref": null,
                  "rendered": "Organization:",
                  "type": "strong",
                },
                " ",
                "org1",
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Resources
</h3>,
              <ul
                className="list-inline"
>
                <li
                                className="list-group-item d-flex align-items-center"
                >
                                <span
                                                className="badge badge-secondary"
                                >
                                                CSV
                                </span>
                                <a
                                                className="px-3"
                                                href={undefined}
                                >
                                                rsc1
                                </a>
                                <span
                                                className="ml-auto mr-3"
                                >
                                                Wed, 20 Jun 2018 22:00:00 GMT
                                </span>
                                <span
                                                className="badge badge-pill badge-dark"
                                >
                                                public
                                </span>
                </li>
</ul>,
            ],
            "className": "my-4",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Resources",
              },
              "ref": null,
              "rendered": "Resources",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    className="list-group-item d-flex align-items-center"
>
                    <span
                                        className="badge badge-secondary"
                    >
                                        CSV
                    </span>
                    <a
                                        className="px-3"
                                        href={undefined}
                    >
                                        rsc1
                    </a>
                    <span
                                        className="ml-auto mr-3"
                    >
                                        Wed, 20 Jun 2018 22:00:00 GMT
                    </span>
                    <span
                                        className="badge badge-pill badge-dark"
                    >
                                        public
                    </span>
</li>,
                ],
                "className": "list-inline",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span
                        className="badge badge-secondary"
>
                        CSV
</span>,
                      <a
                        className="px-3"
                        href={undefined}
>
                        rsc1
</a>,
                      <span
                        className="ml-auto mr-3"
>
                        Wed, 20 Jun 2018 22:00:00 GMT
</span>,
                      <span
                        className="badge badge-pill badge-dark"
>
                        public
</span>,
                    ],
                    "className": "list-group-item d-flex align-items-center",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "CSV",
                        "className": "badge badge-secondary",
                      },
                      "ref": null,
                      "rendered": "CSV",
                      "type": "span",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "rsc1",
                        "className": "px-3",
                        "href": undefined,
                      },
                      "ref": null,
                      "rendered": "rsc1",
                      "type": "a",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Wed, 20 Jun 2018 22:00:00 GMT",
                        "className": "ml-auto mr-3",
                      },
                      "ref": null,
                      "rendered": "Wed, 20 Jun 2018 22:00:00 GMT",
                      "type": "span",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "public",
                        "className": "badge badge-pill badge-dark",
                      },
                      "ref": null,
                      "rendered": "public",
                      "type": "span",
                    },
                  ],
                  "type": "li",
                },
              ],
              "type": "ul",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="material-icons mr-1"
>
                open_in_new
</i>,
              " View on Datasud.fr",
            ],
            "className": "btn btn-success mb-1",
            "href": "https://trouver.datasud.fr/dataset/undefined",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "open_in_new",
                "className": "material-icons mr-1",
              },
              "ref": null,
              "rendered": "open_in_new",
              "type": "i",
            },
            " View on Datasud.fr",
          ],
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
